test_common/harness/ThreadPool.cpp

```diff
@@ -634,6 +641,7 @@ void ThreadPool_Init(void)
         pthread_t tid = 0;
         err = pthread_create(&tid, NULL, ThreadPool_WorkerFunc,
                              (void *)&threadID);
+        pthreads.push_back(tid);
```

Comment:
I'd suggest to do this after checking `err` to avoid storing garbage into the vector in case `pthread_create` fails.

Reply:
Good spot.

Reply:
Fixed.